improvement: add Anthos GKE on AWS / Azure Terraform (#124)
GoogleCloudPlatform/anthos-samples: anthos-multi-cloud/AWS/README.md

> # GKE on AWS Terraform

Reviewer: ```suggestion
# Install GKE on AWS using Terraform
```

> ## Notes:

Reviewer: ```suggestion
```

Reviewer: Remove this notes part

> # GKE on AWS Terraform
> 
> ## Notes:
> ![Anthos Multi-Cloud](Anthos-Multi-AWS.png)
> 
> This terraform script will install all relevant [IaaS prerequisites](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/prerequisites) in AWS(VPC, , subnets, internet gateay, NAT gateway, IAM roles, route tables, and KMS) and then deploy Anthos GKE with 3 control plane nodes(1 in each AZ) of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/) and a single node pool of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/)  with 2 nodes in an autoscaling group to max 5 nodes to the AWS us-east-1 region. The node pool will be deployed to the us-east-1a zone. The network topology setup is documented [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/create-aws-vpc#create-sample-vpc).  You can adjust the region and AZs in the variables.tf file. For a list of AWS regions and associated K8s version supported per GCP region please use this command:
> 

Reviewer: ```suggestion
# Install GKE on AWS using Terraform

This script is meant to be a quick start to working with Anthos on AWS. For more information on Anthos Multi-Cloud please [click here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/). This terraform script will install all relevant [IaaS prerequisites](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/prerequisites) in AWS _(VPC, , Subnets, Internet gateway, NAT gateway, IAM Roles, Route tables, and KMS)_.

![Anthos Multi-Cloud](Anthos-Multi-AWS.png)

 **The Terraform script deploys Anthos GKE with:**
- 3 control plane nodes _(1 in each AZ)_ of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/).
- A single node pool of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/) with 2 nodes in an autoscaling group to max 5 nodes to the AWS `us-east-1` region. 

**Other information:**
- The node pool will be deployed to the `us-east-1a` zone. 
- Supported instance types in AWS can be found [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/reference/supported-instance-types).
- The network topology setup is documented [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/create-aws-vpc#create-sample-vpc).
- You can adjust the region and AZs in the [variables.tf](/anthos-multi-cloud/AWS/variables.tf) file. 
- For a list of AWS regions and associated K8s version supported per GCP region please use the following command:
```

> # GKE on AWS Terraform
> 
> ## Notes:
> ![Anthos Multi-Cloud](Anthos-Multi-AWS.png)
> 
> This terraform script will install all relevant [IaaS prerequisites](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/prerequisites) in AWS(VPC, , subnets, internet gateay, NAT gateway, IAM roles, route tables, and KMS) and then deploy Anthos GKE with 3 control plane nodes(1 in each AZ) of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/) and a single node pool of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/)  with 2 nodes in an autoscaling group to max 5 nodes to the AWS us-east-1 region. The node pool will be deployed to the us-east-1a zone. The network topology setup is documented [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/create-aws-vpc#create-sample-vpc).  You can adjust the region and AZs in the variables.tf file. For a list of AWS regions and associated K8s version supported per GCP region please use this command:
> 
> ```bash
> gcloud container aws get-server-config --location [gcp-region]
> ```
>  Supported instance types in AWS can be found [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/reference/supported-instance-types). After the cluster has been installed it will show up in your GKE page of the GCP console in your relevant GCP project.
> 
>  This script is meant to be a quick start to working with Anthos on AWS. For more information on Anthos Multi-Cloud please [click here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/).

Reviewer: ```suggestion
# Install GKE on AWS using Terraform

This script is meant to be a quick start to working with Anthos on AWS. For more information on Anthos Multi-Cloud please [click here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/). This terraform script will install all relevant [IaaS prerequisites](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/prerequisites) in AWS _(VPC, , Subnets, Internet gateway, NAT gateway, IAM Roles, Route tables, and KMS)_.

![Anthos Multi-Cloud](Anthos-Multi-AWS.png)

 **The Terraform script deploys Anthos GKE with:**
- 3 control plane nodes _(1 in each AZ)_ of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/).
- A single node pool of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/) with 2 nodes in an autoscaling group to max 5 nodes to the AWS `us-east-1` region. 

**Other information:**
- The node pool will be deployed to the `us-east-1a` zone. 
- Supported instance types in AWS can be found [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/reference/supported-instance-types).
- The network topology setup is documented [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/create-aws-vpc#create-sample-vpc).
- You can adjust the region and AZs in the [variables.tf](/anthos-multi-cloud/AWS/variables.tf) file. 
- For a list of AWS regions and associated K8s version supported per GCP region please use the following command:
  ```bash
  gcloud container aws get-server-config --location [gcp-region]
  ``` 

After the cluster has been installed it will show up in the [Kubernetes Engine page](https://console.cloud.google.com/kubernetes/list/overview) of the GCP console in your relevant GCP project. 
```

# Install GKE on AWS using Terraform

This script is meant to be a quick start to working with Anthos on AWS. For more information on Anthos Multi-Cloud please [click here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/). This terraform script will install all relevant [IaaS prerequisites](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/prerequisites) in AWS _(VPC, , Subnets, Internet gateway, NAT gateway, IAM Roles, Route tables, and KMS)_.

![Anthos Multi-Cloud](Anthos-Multi-AWS.png)

 **The Terraform script deploys Anthos GKE with:**
- 3 control plane nodes _(1 in each AZ)_ of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/).

> - A single node pool of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/) with 2 nodes in an autoscaling group to max 5 nodes to the AWS `us-east-1` region. 

Reviewer: ```suggestion
- A single node pool of type [t3.medium](https://aws.amazon.com/ec2/instance-types/t3/) with 2 nodes in an autoscaling group to max 5 nodes to the AWS `us-east-1` region.
```

**Other information:**

> - The node pool will be deployed to the `us-east-1a` zone. 

Reviewer: ```suggestion
- The node pool will be deployed to the `us-east-1a` zone.
```

- Supported instance types in AWS can be found [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/reference/supported-instance-types).
- The network topology setup is documented [here](https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/create-aws-vpc#create-sample-vpc).

> - You can adjust the region and AZs in the [variables.tf](/anthos-multi-cloud/AWS/variables.tf) file. 

Reviewer: ```suggestion
- You can adjust the region and AZs in the [variables.tf](/anthos-multi-cloud/AWS/variables.tf) file.
```
